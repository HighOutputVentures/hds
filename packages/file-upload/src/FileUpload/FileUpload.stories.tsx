import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import FileUpload from './FileUpload';
import { Flex, ThemeProvider } from '@highoutput/hds';

export default {
  title: 'Components/File Upload/File Upload',
  component: FileUpload,
} as ComponentMeta<typeof FileUpload>;

const Template: ComponentStory<typeof FileUpload> = (args) => (
  <ThemeProvider>
    <Flex justify="center">
      <FileUpload {...args} />
    </Flex>
  </ThemeProvider>
);

export const Default = Template.bind({});

Default.args = {
  ...Default.args,
  accepts: '.png, .gif, .jpeg, .svg',
};
