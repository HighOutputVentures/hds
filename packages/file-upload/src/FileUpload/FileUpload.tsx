import { Box, Flex, Icon, Input, Stack, Text } from '@chakra-ui/react';
import React, { FC } from 'react';
import { FileUploadIcon } from '@highoutput/hds-icons';
import { ChangeHandler } from 'react-hook-form';

const color = {
  bg: '#E5E7EB',
  textColor: { gray: '#6B7280', purple: '#6B46C1' },
};

export interface FileUpload {
  accepts?: string;
  onChange?: ChangeHandler;
  onBlur?: ChangeHandler;
  disabled?: boolean;
  __testId?: string;
}

const FileUpload: FC<FileUpload> = ({
  accepts,
  onChange,
  __testId,
  disabled,
  onBlur,
}) => {
  return (
    <>
      <Flex
        direction="column"
        h="8rem"
        w="32rem"
        border="2px"
        borderColor={color.bg}
        borderRadius="0.5rem"
        pos="relative"
        justifyContent="center"
      >
        <Box display="flex" justifyContent="center" alignItems="center" mb="0.75rem">
          <Box
            display="flex"
            justifyContent="center"
            alignItems="center"
            borderRadius="50%"
            h="2.5rem"
            w="2.5rem"
            bg={color.bg}
          >
            <Icon
              color={color.textColor.gray}
              h="1.5rem"
              w="1.5rem"
              as={FileUploadIcon}
            />
          </Box>
        </Box>
        <Stack spacing={1} direction="row" justifyContent="center">
          <Text color={color.textColor.purple} size="button-default">
            Click to upload
          </Text>
          <Text color={color.textColor.gray} size="paragraph-xs-default">
            or drag and drop
          </Text>
        </Stack>
        <Text
          textAlign="center"
          size="paragraph-xxs-default"
          color={color.textColor.gray}
        >
          {accepts?.toLocaleUpperCase()} (max. 800x400px)
        </Text>
        <Input
          id="file"
          type="file"
          w="100%"
          h="100%"
          opacity={0}
          pos="absolute"
          data-testid={__testId ?? 'hds.file.input.upload'}
          top={0}
          cursor="pointer"
          accept={accepts}
          objectPosition={{
            base: 'center center',
            lg: 'initial',
          }}
          objectFit={{ base: 'cover', lg: 'initial' }}
          onChange={onChange}
          onBlur={onBlur}
          disabled={disabled}
        />
      </Flex>
    </>
  );
};

export default FileUpload;
