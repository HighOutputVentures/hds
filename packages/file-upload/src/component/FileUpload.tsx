import { Box, Flex, Icon, Input, Stack, Text } from '@chakra-ui/react';
import React, { FC } from 'react';

const FileUploadIcon = require('../../../../assets/icon/file-upload-icon.svg');

const FileUpload: FC = () => {
  return (
    <>
      <Flex
        direction="column"
        h="8rem"
        w="32rem"
        border="2px"
        borderColor="#E5E7EB"
        borderRadius="0.5rem"
        pos="relative"
        justifyContent="center"
      >
        <Box display="flex" justifyContent="center">
          <Icon color="#6B7280" h="2.25rem" w="2.25rem" as={FileUploadIcon} />
        </Box>
        <Stack
          spacing={1}
          direction="row"
          fontSize="0.875rem"
          lineHeight="1.25rem"
          fontWeight={500}
          justifyContent="center"
        >
          <Text color="#6B46C1" size="button-default">
            Click to upload
          </Text>
          <Text color="#6B7280" size="paragraph-xs-default">
            or drag and drop
          </Text>
        </Stack>
        <Text textAlign="center" size="paragraph-xxs-default" color="#6B7280">
          SVG, PNG, JPG or GIF (max. 800x400px)
        </Text>
        <Input
          id="file"
          type="file"
          w="100%"
          h="100%"
          opacity={0}
          pos="absolute"
          top={0}
          cursor="pointer"
          accept=".png, .gif, .jpeg, .svg"
          objectPosition={{
            base: 'center center',
            lg: 'initial',
          }}
          objectFit={{ base: 'cover', lg: 'initial' }}
        />
      </Flex>
    </>
  );
};

export default FileUpload;
