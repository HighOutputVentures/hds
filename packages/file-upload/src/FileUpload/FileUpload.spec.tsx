import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import FileUpload from './FileUpload';

describe('FileUpload', () => {
  it('should render FileUpload', () => {
    const { container } = render(<FileUpload />);

    expect(container).toMatchSnapshot();
  });

  it('should have a input type of file upload', () => {
    const { getByTestId } = render(<FileUpload />);

    const input = getByTestId('hds.file.input.upload');
    expect(input).toHaveProperty('type', 'file');
  });
});
