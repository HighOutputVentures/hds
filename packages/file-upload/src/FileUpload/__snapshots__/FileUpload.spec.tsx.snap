// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload should render FileUpload 1`] = `
<div>
  <div
    class="css-n1dzwc"
  >
    <div
      class="css-11cup7y"
    >
      <div
        class="css-19thdhd"
      >
        <svg
          class="chakra-icon chakra-icon css-wfawus"
          focusable="false"
          viewBox="0 0 20 20"
        >
          <path
            d="M6.66675 13.3333L10.0001 10M10.0001 10L13.3334 13.3333M10.0001 10V17.5M16.6667 13.9524C17.6847 13.1117 18.3334 11.8399 18.3334 10.4167C18.3334 7.88536 16.2814 5.83333 13.7501 5.83333C13.568 5.83333 13.3976 5.73833 13.3052 5.58145C12.2185 3.73736 10.2121 2.5 7.91675 2.5C4.46497 2.5 1.66675 5.29822 1.66675 8.75C1.66675 10.4718 2.36295 12.0309 3.48921 13.1613"
            stroke="#2E2E2E"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.66667"
          />
        </svg>
      </div>
    </div>
    <div
      class="chakra-stack css-bdrdb0"
    >
      <p
        class="chakra-text css-3lx6gy"
      >
        Click to upload
      </p>
      <p
        class="chakra-text css-1azrnsv"
      >
        or drag and drop
      </p>
    </div>
    <p
      class="chakra-text css-ojejxq"
    >
       (max. 800x400px)
    </p>
    <input
      class="chakra-input css-167pywf"
      data-testid="hds.file.input.upload"
      id="file"
      type="file"
    />
  </div>
</div>
`;
